### ISSUE TITLE
Nested Objects and default values

### ISSUE THREAD

**Issue by matteobattista** [dynamoosejs/dynamoose#527]
### Summary:

Something wrong with nested object default values... below my schema, expected behavior and unexpected.

#### Schema
```
const schemaAttributes =   {
  id: { type: String, hashKey: true },
  cover: { type: String, required: true },
  categories: [{ type: String, required: true }],
  titles: [{
    lang: { type: String, enum: ['it', 'en', 'fr', 'de'], required: true },
    value: { type: String, required: true }
  }],
  results: [{
    gender: { type: String, enum: ['m', 'f', 'u'], required: true},
    style: {
      description: {
        align: { type: String, enum: ['left', 'center', 'right'], default: 'center' },
        color: { type: String, default: '#333' },
        font: { type: String, default: '24px Arial' },
        position: {
          x: { type: Number, default: 450 },
          y: { type: Number, default: 200 }
        },
        width: { type: Number, default: 350 }
      },
      picture: {
        size: { type: Number, required: true, default: 250 }
      }
    }
  }]
}
```

#### General
Create object:
```
{ 
  "cover": "/path/to/image0.png",
  "categories": ["cat1", "cat2"],
  "titles":[{
    "lang": "en",
    "value": "Are you ok?"
  }],
  "results": [{
    "gender": "u",
    "titles":[{
      "lang": "en",
      "value": "Yes"
    }],
    "style": {
      "description": {
        "align": "center"
      }
    }
  }]
}
```

### Current output and behavior:
```
{
    "id": "390cd913-6fa2-4302-a02b-4b314a414074",
    "cover": "/path/to/image10.png",
    "categories": [
        "cat1",
        "cat2"
     ],
    "titles": [
        {
            "lang": "en",
           "value": "Are you ok?"
        }
    ],
    "results": [
        {
            "gender": "u",
            "titles": [
                {
                    "lang": "en",
                    "value": "Yes"
                }
            ],
            "style": {
                "description": {
                    "align": "center"
                }
            }
        }
    ]
}
```

### Expected output and behavior:
```
{
    "id": "390cd913-6fa2-4302-a02b-4b314a414074",
    "cover": "/path/to/image10.png",
    "categories": [
        "cat1",
        "cat2"
     ],
    "titles": [
        {
            "lang": "en",
           "value": "Are you ok?"
        }
    ],
    "results": [
        {
            "gender": "u",
            "titles": [
                {
                    "lang": "en",
                    "value": "Yes"
                }
            ],
            "style": {
                "description": {
                    "align": "center",
                    "color": "#333",
                    "font": "24px Arial",
                    "position": {
                       "x": 450,
                       "y": 200
                    },
                    "width": 350
                }
            }
        }
    ]
}
```

### Environment:

Operating System: macOs Mojave
Operating System Version: 10.14.2
Node.js version (`node -v`): 11.3.0
NPM version: (`npm -v`): 6.5.0
Dynamoose version: 1.3.0
Dynamoose Plugins:

-

### Dynamoose Plugins:

- [ ] Yes, I believe that one or multiple 3rd party Dynamoose plugins are effecting this issue
- [ ] No, I believe this issue is independent of any 3rd party Dynamoose plugins I'm using
- [x] Unknown, I'm unsure if Dynamoose plugins are effecting this issue
- [ ] I am not using any Dynamoose plugins

### Type (select 1):
- [X] Bug report
- [ ] Feature suggestion
- [ ] Question
- [ ] Other suggestion
- [ ] Something not listed here

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [ ] I have ensured that all of my plugins that I'm using with Dynamoose are listed above
- [ ] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by fishcharlie** [comment]
@matteobattista I'm confused about what the problem is. How are you creating the object? How are you retrieving the object? What is the problem? Lot more detail needed in order to help.

**Comment by matteobattista** [comment]
Hi @fishcharlie, I'll try to explain me better... Schema is clear? The object is passed via http POST to create new item in dynamodb... and what I expect is that default value are insert in the item also I don't insert them in the object I send by http POST... I hope I have been more clear now.
Thanks for the support

**Comment by fishcharlie** [comment]
@mattebattista No. That doesn’t help at all. Your web server is irrelevant to me. Dynamoose doesn’t relate at all to your web server. If that is your problem, then that isn’t a Dynamoose problem. I’m only interested in your Dynamoose setup, and you have no further information about that.

Please open a new PR that adds a failing test case that represents your problem to the test suite. This test should not include any web server code and should represent your problem in the most minimal way possible. Once that is complete post the PR number here.

**Comment by matteobattista** [comment]
I'll try to simplify the problem... "default" attribute doens't work on nested object...
Maybe you can refer to this old issue: https://github.com/dynamoosejs/dynamoose/issues/267

**Comment by fishcharlie** [comment]
@matteobattista Sorry but I have to close this issue. I have to be picky about my time and prioritize things. If I can't get support about issues I have to close them.

I asked you to

> Please open a new PR that adds a failing test case that represents your problem to the test suite. This test should not include any web server code and should represent your problem in the most minimal way possible. Once that is complete post the PR number here.

Once that is complete I will consider reopening this issue.

**Comment by chrisbrantley** [comment]
I also have the same problem as @matteobattista: Defaults are not working for nested objects. I will try to get a failing test added in a PR.
